modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/api/construction/ConstructionPayloads.scala

```diff
@@ -0,0 +1,43 @@
+package org.tessellation.rosetta.domain.api.construction
+
+import cats.data.NonEmptyList
+
+import org.tessellation.ext.derevo.magnoliaCustomizable.snakeCaseConfiguration
+import org.tessellation.rosetta.domain.api.construction.ConstructionMetadata.MetadataResult
+import org.tessellation.rosetta.domain.operation.Operation
+import org.tessellation.rosetta.domain.{NetworkIdentifier, SigningPayload}
+import org.tessellation.security.hex.Hex
+
+import derevo.cats.{eqv, show}
+import derevo.circe.magnolia.{customizableDecoder, customizableEncoder}
+import derevo.derive
+
+case object ConstructionPayloads {
+
+  @derive(customizableDecoder)
+  case class Request(
+    networkIdentifier: NetworkIdentifier,
+    operations: NonEmptyList[Operation],
+    metadata: MetadataResult
+  )
+
+  @derive(customizableEncoder, eqv, show)
+  case class Response(
+    unsignedTransaction: Hex,
+    payloads: NonEmptyList[SigningPayload]
+  )
+
+  object Response {
+    def fromPayloadsResult(result: PayloadsResult): Response = Response(
+      result.unsignedTransaction,
+      result.payloads
+    )
+  }
+
+  @derive(eqv, show)
+  case class PayloadsResult(
```

Comment:
If `PayloadsResult` is the same as `Response` can't we use only one of these and remove the other?

Reply:
We can remove it but I don't want to return the response directly from the service (which should be agnostic). If you can propose what should be the return type of the service that could be reused for different usages (like for CLI or something), then I'm good for removing that.

Reply:
@sdavidp looks like you resolved this one as well, but this was addressed to @TheMMaciek

Reply:
If we think there is a benefit having these two classes separate then I'm ok with it. I just wanted to clarify because for main protocol we don't have such a separation. Rosetta may require a different approach though so if it makes sense in this case then let's have the separation?

Reply:
A benefit I see in keeping them separate is we can return meaningful types from the service layer which can then be mapped to the appropriate representation for the response. One example that might be improved is [combineTransaction](https://github.com/Constellation-Labs/tessellation/blob/develop/modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/construction/ConstructionService.scala#L112). It returns `Hex` because that's what the response wants but a better type for this result might be something like `case class SerializedSignedTransaction(hex: Hex)`